cpp/src/com/google/idea/blaze/cpp/BlazeConfigurationResolver.java

```diff
@@ -98,9 +100,27 @@ public BlazeConfigurationResolverResult update(
         HeaderRootTrimmer.getValidRoots(
             context, blazeProjectData, toolchainLookupMap, targetFilter, executionRootPathResolver);
     builder.setValidHeaderRoots(validHeaderRoots);
+    builder.setTargetToVersionMap(targetToVersion);
     return builder.build();
   }
 
+  @NotNull
+  private static ImmutableMap<TargetKey, String> getTargetToVersionMap(ImmutableMap<TargetKey, CToolchainIdeInfo> toolchainLookupMap, ImmutableMap<CToolchainIdeInfo, BlazeCompilerSettings> compilerSettings) {
+    Map<ExecutionRootPath, String> compilerVersionByPath =
+            compilerSettings.entrySet().stream().collect(
+                    Collectors.toMap(
+                            e -> e.getKey().getCppExecutable(),
+                            e -> e.getValue().getCompilerVersion()));
+    Map<TargetKey, String> map = toolchainLookupMap.entrySet().stream()
+            .map(e -> new AbstractMap.SimpleImmutableEntry<>(
+                    e.getKey(),
+                    compilerVersionByPath.get(e.getValue().getCppExecutable())))
+            // In case of a broken compiler, the version string is null, but Collectors.toMap requires non-null value function.
+            .filter(e -> e.getValue() != null)
+            .collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
```

Comment:
I think we can use `.collect(toImmutableList())` to avoid `ImmutableList.copyOf()` at the end of the method

`toImmutableList()` can be imported as `import static com.google.common.collect.ImmutableList.toImmutableList;
`